### DISCUSSION TITLE
Overloads crowd out documentation in IntelliSense

### DISCUSSION THREAD

**Discussion by mattbierner** [microsoft/pylance-release#2577] [Enhancement]
Issue Type: <b>Bug</b>

**Repro**
1. In a python file:

```py
import re
```

2. Type `re.sub` and look at IntelliSense 

**Bug**
The list of 4 overloads for `re.sub` crowds out the arguably much more useful documentation on what this function does

<img width="997" alt="Screen Shot 2021-04-02 at 12 55 11 AM" src="https://user-images.githubusercontent.com/212305/113395884-7041ec00-934f-11eb-848e-42d0caeac896.png">

Compared this with a similar overloaded function in JS/TS, which lists the common overload first and then indicates that additional overloads are available:

<img width="1030" alt="Screen Shot 2021-04-02 at 1 05 50 AM" src="https://user-images.githubusercontent.com/212305/113395975-94053200-934f-11eb-8899-b624eaad5c87.png">

Extension version: 2021.3.4
VS Code version: Code - Insiders 1.55.0-insider (05ce54df77c242b1b8ea5fd93d720c62fdb183e6, 2021-03-25T05:12:59.457Z)
OS version: Darwin x64 20.3.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700K CPU @ 4.20GHz (8 x 4200)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off_ok<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 3|
|Memory (System)|40.00GB (0.03GB free)|
|Process Argv|--crash-reporter-id cefe23df-690f-4b1b-9ce9-e004d3be2745|
|Screen Reader|no|
|VM|0%|
</details>

**Comment by mattbierner** [comment]
Discussed in #612. However that fix for that issue was improving the formatting of the overload. This issue is about all these overloads being listed at the top of IntelliSense

**Comment by mattbierner** [comment]
Another example for the `open` function:

<img width="1013" alt="Screen Shot 2021-04-03 at 6 47 07 PM" src="https://user-images.githubusercontent.com/212305/113496295-0ccdd080-94ad-11eb-8231-52240532829e.png">

**Comment by heejaechang** [comment]
@judej we need decision on this. I think we talked about it long time ago but didn't decide. and I think Savannah did some user study on this as well?

**Comment by judej** [comment]
Converting this issue to an enhancement discussion for comments and upvotes
